### ISSUE TITLE
Why l_d_real have to multiply 0.5

### ISSUE THREAD

**Issue by zpkosmos** [open-mmlab/mmsr#23]
Thanks for your great work!
I have some question about the following line in your code
https://github.com/open-mmlab/mmsr/blob/8c615f07638e55003dcacd0808b711d62241b20b/codes/models/SRGAN_model.py#L182
In the code，the l_d_real multiply 0.5 after calculate the loss，and pred_d_fake is defined twice, what is the different bewteen your final code and the code you have commented out
https://github.com/open-mmlab/mmsr/blob/8c615f07638e55003dcacd0808b711d62241b20b/codes/models/SRGAN_model.py#L183
pred_d_real = self.netD(self.var_ref)
pred_d_fake = self.netD(self.fake_H.detach())  # detach to avoid BP to G
l_d_real = self.cri_gan(pred_d_real - torch.mean(pred_d_fake), True)
l_d_fake = self.cri_gan(pred_d_fake - torch.mean(pred_d_real), False)
l_d_total = (l_d_real + l_d_fake) / 2
l_d_total.backward()
